server/market/market.go

```diff
@@ -349,26 +448,74 @@ func (m *Market) processOrder(rec *orderRecord, epoch *EpochQueue, errChan chan<
 	go m.notifyNewOrder(ord, epochAction)
 }
 
-// processEpoch performs the following operations for a closed epoch. The
-// EpochQueue's Orders map must not be modified by another goroutine:
+func (m *Market) unlockCancelOrderCoins(co *order.CancelOrder) {
+	oid := co.TargetOrderID
+	// The OrderID does not indicate the canceled order's side (or asset), try
+	// the ID in both base and quote coin lockers.
```

Comment:
You can get the side and assets from the `CancelOrder`, which has a `Prefix`. The data in the `Prefix` was validated by the `OrderRouter` and by the `MarketTunnel`'s `Cancelable` method, once implemented.

Reply:
This was supposed to have been removed.  It was for a quick test.  However, for a CancelOrder, the Prefix doesn't indicate which side the TargetOrderID is on.  Turns out this was not needed though.  The `Matcher` deals with cancel orders and all we care about was the `unbooked` slice returned by `Match`.

Reply:
Oh yeah. Forgot the prefix doesn't have the side.